app/api/routes.py

```diff
@@ -179,6 +184,32 @@ def get_resources():
     return standardize_response(payload=dict(data=resource_list, **pagination_details))
 
 
+def search_results():
+    term = request.args.get('q', '', str)
+    page = request.args.get('page', 0, int)
+    query = Resource.query
+
+    search_result = index.search(f'{term}', {
+        'page': page,
+    })
+
+    if page >= int(search_result['nbPages']):
+        return redirect('/404')
+
+    object_ids = [result['objectID'] for result in search_result['hits']]
+    results = [r.serialize for r in query.filter(Resource.id.in_(object_ids))]
+
+    pagination_details = {
+                "pagination_details": {
+                    "page": search_result['page'],
+                    "number_of_pages": search_result['nbPages'],
+                    "records_per_page": search_result['hitsPerPage'],
+                    "total_count": search_result['nbHits'],
+                }
+        }
+    return standardize_response(payload=dict(data=results, **pagination_details))
```

Comment:
We need some slightly better error handling here. If there's an internal server error, you get something similar to the following:
```json
{
    "apiVersion": "1.0",
    "data": null,
    "errors": [
        {
            "code": "something-went-wrong"
        }
    ],
    "number_of_pages": 0,
    "page": 0,
    "records_per_page": 20,
    "status": "error",
    "status_code": 500,
    "total_count": 0
}
```

Reply:
Perhaps the solution to this can go inside `standardize_response` but I'll let you take that in whichever direction you want